sensor/tests/resource/pod/pod_test.go

```diff
@@ -0,0 +1,93 @@
+package pod
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stackrox/rox/sensor/tests/resource"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	appsv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+)
+
+type PodHierarchySuite struct {
+	testContext *resource.TestContext
+	suite.Suite
+}
+
+func Test_PodHierarchy(t *testing.T) {
+	suite.Run(t, new(PodHierarchySuite))
+}
+
+var _ suite.SetupAllSuite = &PodHierarchySuite{}
+
+func (s *PodHierarchySuite) SetupSuite() {
+	if testContext, err := resource.NewContext(s.T()); err != nil {
+		s.Fail("failed to setup test context: %s", err)
+	} else {
+		s.testContext = testContext
+	}
+}
+
+func (s *PodHierarchySuite) Test_ContainerSpecOnDeployment() {
+	s.testContext.RunWithResources([]resource.YamlTestFile{
+		resource.NginxDeployment,
+		resource.NginxServiceAccount,
```

Comment:
Question, is this only necessary because the `nginx.yaml` has a `serviceAccountName` used in the role tests? Do you think it would make sense to have another `nginx.yaml` without the `serviceAccountName` so the tests don't create an extra resources?

Reply:
Replied in the top level comment.